R/write.magpie.R

```diff
@@ -119,8 +119,9 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
     }
 
     # look for comment/additional information
-    if (is.null(comment) & !is.null(attr(x, "comment"))) comment <- attr(x, "comment")
+    if (is.null(comment) && !is.null(attr(x, "comment"))) comment <- attr(x, "comment")
```

Comment:
what is the effect of the additional `&`?

Reply:
As far as I understand `&` can do vector comparisons and you might end up with a vector of TRUE and FALSE statements. `&&` can only compare single logical values but has the advantage that the second statement is not being evaluated if the first statement is already FALSE. I changed it in the code as my lintr version was complaining about and the `&&` and `||` is indeed the better version here while it probably does not make a big difference here what to take